apps/files/src/components/Trashbin.vue

```diff
@@ -123,26 +131,45 @@ export default {
     },
 
     $_ocTrashbin_clearTrashbinConfirmation (files = this.selectedFiles) {
-      for (const file of files) {
-        this.$client.fileTrash.clearTrashBin(file.id)
-          .then(() => {
-            this.$_ocTrashbin_removeFileFromList([file])
-            const translated = this.$gettext('%{file} was successfully deleted')
-            this.showMessage({
-              title: this.$gettextInterpolate(translated, { file: file.name }, true)
+      // TODO: use clear all if all files are selected
+      const deleteOps = []
+
+      const self = this
+      function deleteFile (file) {
+        return async () => {
+          return self.$client.fileTrash.clearTrashBin(file.id)
+            .then(() => {
+              self.$_ocTrashbin_removeFileFromList([file])
+              const translated = self.$gettext('%{file} was successfully deleted')
+              self.showMessage({
+                title: self.$gettextInterpolate(translated, { file: file.name }, true)
+              })
             })
-          })
-          .catch(error => {
-            const translated = this.$gettext('Deletion of %{file} failed')
-            this.showMessage({
-              title: this.$gettextInterpolate(translated, { file: file.name }, true),
-              desc: error.message,
-              status: 'danger'
+            .catch(error => {
+              if (error.statusCode && error.statusCode === 423) {
```

Comment:
Does here need to be the first condition? Don't we already know if the statusCode exists if the second condition is true?